api/src/main/java/org/apache/iceberg/actions/ActionsProvider.java

```diff
@@ -74,4 +74,11 @@ default ExpireSnapshots expireSnapshots(Table table) {
   default DeleteReachableFiles deleteReachableFiles(String metadataLocation) {
     throw new UnsupportedOperationException(this.getClass().getName() + " does not implement deleteReachableFiles");
   }
+
+  /**
+   * Instantiates an action to generate CDC records.
+   */
+  default Cdc generateCdcRecords(Table table) {
```

Comment:
Nit: Consider using `CDC` as all caps instead, like it is in the javadoc comments. For me, it looks a lot cleaner.

Reply:
I'm also open to other names, e.g., `ChangeDataSet`, `ChangeDataCapture`

Reply:
In Flink it's referred to as `Changelog`.

Reply:
Yeah I agree that using a full word would be better.

In comments and even method names it's fine in my opinion but as the main class name it probably would be best to use the full name.

Reply:
+1 for `Changelog`. Here, it means `generateChangelog`

Reply:
I am OK with `generateChangelog`.

Reply:
or `generateChangeSet` for batch execution. if it is a long-running streaming execution, `changelog` would be more accurate as it implies a stream.

Reply:
I liked `generateChangelog` but if it can confuse people, `generateChangeSet` sounds good too.

Reply:
Combining feedbacks, changed it to `GetChangeSet`. The name `GenerateChangeSet` is good, but it is way too long. Think about the class name `BaseGenerateChangeSetSparkActionResult`. I admit the verb `get` is plain comparing to `generate`. But I think it is fine, a plain name is suitable for a tool.